### ISSUE TITLE
Remove Lightning dependency

### ISSUE THREAD

**Issue by wrongu** [bonsai-neuro-ai/nn-library#4]
In an earlier version of `nn-library` I thought we would want to lean on pytorch lightning to accelerate things. That turned out to be more headache than helpful. Removing it will require some work:

- refactor the class hierarchy in all classes defined in `src/nn_lib/datasets/`, which currently derive from our custom `TorchvisionDataModuleBase` which derives from `LightningDataModule`. This was too many layers of inheritance/abstraction for what is meant to be a low-complexity way for all of us to use/reuse the same datasets. Key features that we want from some data abstraction:
    - good default configuration. If two people instantiate `ImageNet` then the default options should make it behave the same for both users, and those defaults should have good loading performance. For example, the systems we typically work on are fastest when DataLoaders are given `num_workers=4` (plus or minus) and `pin_memory=True`. These should be the defaults in our system.
    - don't duplicate preprocessing. For instance, z-score normalizing pixels should only have to be calculated once and then shared with other users on the same system, saving the relevant metadata in some standard place relative to the dataset root directory
    - we should prepare to migrate to FFCV soon and our refactor should have that in mind, though that will be a separate issue. Computing FFCV .beton files is another example of preprocessing that should just be done once per server in a centralized place
- remove `src/nn_lib/trainer.py` and anything that references it
- remove `src/nn_lib/models/lit_classifier.py` and anything that references it
- check all `env` and `config` portions of the code and remove lighting there
- update demos and documentation
- remove lightning from the list of pyproject.toml dependencies
